cmd/natsdumper/csvwriter/csv_writer.go

```diff
@@ -0,0 +1,197 @@
+)
+
+var src = rand.NewSource(time.Now().UnixNano())
+var sessionID string = randStringBytesMaskImprSrc(16)
+
+type marketOrderStruct struct {
+	Id               int    `json:"Id"`
+	ItemTypeId       string `json:"ItemTypeId"`
+	LocationId       int    `json:"LocationId"`
+	QualityLevel     int    `json:"QualityLevel"`
+	EnchantmentLevel int    `json:"EnchantmentLevel"`
+	UnitPriceSilver  int    `json:"UnitPriceSilver"`
+	Amount           int    `json:"Amount"`
+	AuctionType      string `json:"AuctionType"`
+	Expires          string `json:"Expires"`
+}
+
+type goldPricesUploadStruct struct {
+	Prices     []int `json:"Prices"`
+	Timestamps []int `json:"Timestamps"`
+}
+
+func randStringBytesMaskImprSrc(n int) string {
+	b := make([]byte, n)
+	// A src.Int63() generates 63 random bits, enough for letterIdxMax characters!
+	for i, cache, remain := n-1, src.Int63(), letterIdxMax; i >= 0; {
+		if remain == 0 {
+			cache, remain = src.Int63(), letterIdxMax
+		}
+		if idx := int(cache & letterIdxMask); idx < len(letterBytes) {
+			b[i] = letterBytes[idx]
+			i--
+		}
+		cache >>= letterIdxBits
+		remain--
+	}
+
+	return string(b)
+}
+
+func fileExists(filename string) bool {
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		return false
+	}
+
+	return true
+}
+
+func constructHeader(orderTypeStruct reflect.Type) string {
+	var outHeader string = ""
+	for i := 0; i < orderTypeStruct.NumField(); i++ {
+		outHeader += orderTypeStruct.Field(i).Name
+
+		if i < orderTypeStruct.NumField()-1 {
+			outHeader += ","
+		}
+	}
+
+	return outHeader
+}
+
+func floatToString(input_num float64) string {
+	// to convert a float number to a string
+	return strconv.FormatFloat(input_num, 'f', 6, 64)
+}
+
+func valToString(val interface{}) string {
+	switch val.(type) {
+	case int:
+		return fmt.Sprint(val.(int64))
+	case float64:
+		if val.(float64) == float64(int64(val.(float64))) {
+			return fmt.Sprint(int64(val.(float64)))
+		} else {
+			return floatToString(val.(float64))
+		}
+	case string:
+		return "\"" + fmt.Sprint(val.(string)) + "\""
+	case []interface{}:
+		outStr := "\""
+		v := val.([]interface{})
+		for ii := 0; ii < len(v); ii++ {
+
+			outStr += valToString(v[ii])
+
+			if ii < len(v)-1 {
+				outStr += ","
+			} else {
+				outStr += "\""
+			}
+		}
+		return outStr
+	}
+	return ""
+}
+
+func constructEntryFromJSON(orderData []byte, orderTypeStruct reflect.Type) string {
+	var o map[string]interface{} //map[string]interface{}
+
+	err := json.Unmarshal(orderData, &o)
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		return ""
+	}
+
+	var outStr string = ""
+
+	for i := 0; i < orderTypeStruct.NumField(); i++ {
+		fieldName := orderTypeStruct.Field(i).Name
+
+		fn := o[fieldName]
+		outStr += valToString(fn)
+
+		if i < len(o)-1 {
+			outStr += ","
+		}
+	}
+
+	return outStr
+}
+
+func writeDataToFile(msg *nats.Msg, identification string, filename string, orderTypeStruct reflect.Type) bool {
```

Comment:
Have not used it, but whats wrong with using the built-in CSV writer: https://golang.org/pkg/encoding/csv/